addons/website/static/tests/tours/website_form_editor.js

```diff
@@ -409,6 +421,22 @@ odoo.define('website.tour.form_editor', function (require) {
             // We have to this that way because the input type = hidden.
             extra_trigger: 'form:has(input[name="email_to"][value="test@test.test"])',
         },
+        {
+            content: "Enter in edit mode again",
+            trigger: "a[data-action='edit']",
+        },
+        {
+            content: "Check that we are in edit mode",
+            trigger: "#oe_snippets.o_loaded",
+            run: () => null, // it's a check
+        },
+        ...addCustomField("char", "text", "''", false),
+        ...addCustomField("char", "text", '""', false),
+        ...addCustomField("char", "text", "``", false),
+        {
+            content: "Save the page",
+            trigger: "button[data-action='save']",
```

Comment:
This should be replaced by our `clickOnSave` util which will be in charged of avoiding race condition.
This is especially true here as the click on save is the last part of the test. Not waiting for the edition to be finished will make the test fail sometimes because some requests will still be ongoing while the testing tour mechanism will be killing the browser, ultimately failing on error such as `Futur Not Settle` something like that

Reply:
In fact, my first idea was indeed to use the `clickOnSave` of the util. However, it seems that the only fact to import `const wTourUtils = require("website.tour_utils");` at the beginning of the document leads to a dependency problem. Indeed, when launching the python test `test_tour`, the first test (`website_form_editor_tour`) executes normally but the second (`website_form_editor_tour_submit`) seems to have have dependency problems and never starts; when the code reaches this test, the terminal displays:
```
Evaluate ready code "odoo.__DEBUG__.services['web_tour.tour'].tours['website_form_editor_tour_submit'].ready"
...
Missing dependencies:     ['website.tour_utils', 'web_editor.snippet.editor']
Non loaded modules:       ['@website/../tests/tours/conditional_visibility', 'website.tour.default_shape_gets_palette_colors', 'website.tour.edit_link_popover', 'website.tour.edit_megamenu', '@website/../tests/tours/edit_menus', '@website/../tests/tours/link_tools', '@website/../tests/tours/media_dialog', '@website/../tests/tours/snippet_background_edition', '@website/../tests/tours/snippet_countdown', 'website.tour.snippet_empty_parent_autoremove', 'website.tour.snippet_image_gallery', '@website/../tests/tours/snippet_images_wall', 'website.tour.snippets_all_drag_and_drop', 'website.tour.form_editor', '@website/../tests/tours/website_page_options', '@website/../tests/tours/website_snippets_menu_tabs']
...
AssertionError: False is not true : The ready "odoo.__DEBUG__.services['web_tour.tour'].tours['website_form_editor_tour_submit'].ready" code was always falsy
```
I am not sure to understand why there is such an error actually. Do you see the problem? (Note that, in the end it will not be a problem for this PR because I will not leave the edit mode as you explained in this comment (https://github.com/odoo/odoo/pull/117632#discussion_r1205707400))

Reply:
No I don't have any idea except something like those being in different bundle and one not loaded but shouldn't be the case here.
We can have a look should you face this error again.